dask/dataframe/io/delta/utils.py

```diff
@@ -0,0 +1,82 @@
+import json
+from typing import Union
+
+import pyarrow as pa
+
+
+def schema_from_string(schema_string: str):
+
+    fields = []
+    schema = json.loads(schema_string)
+    for field in schema["fields"]:
+        name = field["name"]
+        type = field["type"]
+        nullable = field["nullable"]
+        metadata = field["metadata"]
+        pa_type = map_type(type)
+
+        fields.append(pa.field(name, pa_type, nullable=nullable, metadata=metadata))
+    return pa.schema(fields)
+
+
+def map_type(input_type: Union[dict, str]):
+
+    simple_type_mapping = {
+        "byte": pa.int8(),
+        "short": pa.int16(),
+        "integer": pa.int32(),
+        "long": pa.int64(),
+        "float": pa.float32(),
+        "double": pa.float64(),
+        "string": pa.string(),
+        "boolean": pa.bool_(),
+        "binary": pa.binary(),
+        "date": pa.date32(),
+        "timestamp": pa.timestamp("ns"),
+    }
+
+    # If type is string, it should be a "simple" datatype
+    if isinstance(input_type, str):
+
+        # map simple data types, that can be directly converted
+        if input_type in simple_type_mapping:
+            pa_type = simple_type_mapping[input_type]
+        else:
+            raise TypeError(
+                f"Got type unsupported {input_type} when trying to parse schema"
+            )
+
+    # nested field needs special handling
+    else:
+        if input_type["type"] == "array":
```

Comment:
Do we expect pyarrow to handle the full set of nested list/map/struct types?

Reply:
I also initially handled only primitive data types, but after referring to the package code [DeltaLakeReader](https://github.com/jeppe742/DeltaLakeReader), added the nested fields as well. What do you think shall I remove those nested types?